skbio/stats/distance/_permdisp.py

```diff
@@ -289,7 +289,7 @@ def _compute_groups(samples, test_type, grouping):
     if test_type == "centroid":
         centroids = samples.groupby("grouping").aggregate("mean")
     elif test_type == "median":
-        centroids = samples.groupby("grouping").apply(_config_med)
+        centroids = samples.groupby("grouping")[samples.columns.to_list()].apply(_config_med)
```

Comment:
After some research I can see that your solution is a generic and backward compatible one. Good job! One thought: `.to_list()` is not needed, is it? Btw, the line is so long that it violates linting rules.

Reply:
Printing the type of revealed that is a pandas.Index object so I used to_list() to convert it to a python list since I needed to pass the columns in the form of an array. I also used documentation from pandas to help me figure this out https://pandas.pydata.org/docs/reference/api/pandas.Index.to_list.html#pandas.Index.to_list

Reply:
Sounds good to me. Now the only thing remaining is to shorten this line. If you have Ruff pre-commit hook installed correctly, when you do `git commit...` it will automatically run and fix the long line.